aiida/tools/importexport/archive/migrations/v10_to_v11.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Migration from v0.10 to v0.11, used by `verdi export migrate` command.
+
+This migration deals with the file repository. In the old version, the
+"""
+import os
+import shutil
+
+from aiida.tools.importexport.archive.common import CacheFolder
+from .utils import verify_metadata_version, update_metadata
+
+
+def migrate_repository(metadata, data, folder):
+    """Migrate the file repository to a disk object store container."""
+    from disk_objectstore import Container
+    from aiida.repository import Repository, File
+    from aiida.repository.backend import DiskObjectStoreRepositoryBackend
+
+    container = Container(os.path.join(folder.get_path(), 'container'))
+    container.init_container()
+    backend = DiskObjectStoreRepositoryBackend(container=container)
+    repository = Repository(backend=backend)
+
+    for values in data.get('export_data', {}).get('Node', {}).values():
+        uuid = values['uuid']
+        dirpath_calc = os.path.join(folder.get_path(), 'nodes', uuid[:2], uuid[2:4], uuid[4:], 'raw_input')
+        dirpath_data = os.path.join(folder.get_path(), 'nodes', uuid[:2], uuid[2:4], uuid[4:], 'path')
+
+        if os.path.isdir(dirpath_calc):
+            dirpath = dirpath_calc
+        elif os.path.isdir(dirpath_data):
+            dirpath = dirpath_data
+        else:
+            raise AssertionError('node repository contains neither `raw_input` nor `path` subfolder.')
+
+        if not os.listdir(dirpath):
+            continue
+
+        repository.put_object_from_tree(dirpath)
+        values['repository_metadata'] = repository.serialize()
+        # Artificially reset the metadata
+        repository._directory = File()  # pylint: disable=protected-access
+
+    container.pack_all_loose(compress=False)
```

Comment:
one should also call `container.clean_storage()` as `pack_all_loose` only packs but leaves still behind also the loose objects. The current implementation results in a file 2x larger. Ideally, since these files are export files (and so it's OK, I think, to spend a few seconds more to have a more 'cleaned-up' file), I would actually call `container.clean_storage(vacuum=True)` that will take a few seconds more but be more but vacuum the SQLite index file.

Reply:
PS: I rerun adding `container.clean_storage(vacuum=True)` and now the export file size is the same.
Actually, the size remained remarkably almost the same (a reduction of only 0.4%) on a 20GB file - this might be a combined effect of the increase because of the SQL index but the decrease of the zip index and some minimal deduplication (maybe) - note that this profile has few (~10'000) files, so it's not a huge one.

Reply:
Ah, another very important thing! One should also do a `container.close()` before starting to zip! Otherwise, currently I'm finding a `container/packs.idx-shm` and `container/packs.idx-wal` file in the `.zip`, that actually might mean that the DB risks to be corrupted.